api/data_refinery_api/views/stats.py

```diff
@@ -58,8 +71,123 @@ def paginate_queryset_response(queryset, request):
+]
+
+
+def add_type(job_json: dict) -> dict:
+    """Get the type for a job based on its name and add that to the job dict."""
+    # A job name is user_stage_NAME_GOES_HERE_..., so we need to
+    # remove the first two underscore-delimited fields, and then what
+    # comes after depends on the name
+    split_name = job_json["jobName"].split("_")[2:]
+
+    # The last field of a job name is always an ID, so strip that
+    split_name = split_name[:-1]
+
+    # If the new last field is numeric, then it must be a RAM amount, so
+    # strip that too. Otherwise, the last field will be part of the job
+    # name for jobs without RAM amounts, so we want to keep it.
+    if split_name[-1].isnumeric():
+        split_name = split_name[:-1]
+
+    job_json["type"] = "_".join(split_name)
+    return job_json
+
+
+def get_jobs_in_queue(batch_job_queue: str) -> list:
+    """Gets all of the information for the jobs in a batch queue."""
+    jobs = []
+    # AWS Batch only returns one status at a time and doesn't provide a `count` or `total`.
+    for status in [*PENDING_STATUSES, "RUNNING"]:
+        list_jobs_dict = batch.list_jobs(jobQueue=batch_job_queue, jobStatus=status)
+
+        jobs.extend(list_jobs_dict["jobSummaryList"])
+
+        while "nextToken" in list_jobs_dict and list_jobs_dict["nextToken"]:
+            list_jobs_dict = batch.list_jobs(
+                jobQueue=batch_job_queue, jobStatus=status, nextToken=list_jobs_dict["nextToken"],
+            )
+            jobs.extend(list_jobs_dict["jobSummaryList"])
+
+    return jobs
+
+
+def get_batch_jobs_breakdown(force=False):
+    data = {}
+
+    if not settings.RUNNING_IN_CLOUD and not force:
+        # XXX: is this the right move here? What *should* we return if we aren't
+        # running in the cloud? Does this break things?
+        return data
+
+    job_queue_lists = {}
+    for queue_name in settings.AWS_BATCH_QUEUE_ALL_NAMES:
+        try:
+            job_queue_lists[queue_name] = list(map(add_type, get_jobs_in_queue(queue_name)))
+        except Exception:
+            logger.exception(f"Could not get jobs for queue {queue_name}")
+            raise
+
+    all_jobs = functools.reduce(lambda acc, x: acc + x, job_queue_lists.values(), [])
+
+    def is_pending(job):
+        return job["status"] in PENDING_STATUSES
+
+    def is_running(job):
+        return job["status"] == "RUNNING"
+
+    def count_occurrences(pred, it):
+        return functools.reduce(lambda acc, x: acc + 1 if pred(x) else acc, it, 0)
+
+    data["pending_jobs"] = count_occurrences(is_pending, all_jobs)
+    data["running_jobs"] = count_occurrences(is_running, all_jobs)
+
+    def get_job_type(job):
+        return job["type"]
+
+    # groupby must be executed on a sorted iterable
+    # ref: https://docs.python.org/3.8/library/itertools.html#itertools.groupby
+    # XXX: in the old code this was filtered by if the job had the key
+    # "ParameterizedJob". Does Batch have an equivalent?
```

Comment:
No, Batch does not because Batch didn't have to create a weird job-that's-not-a-job because Batch's primary purpose is to be a job queue unlike an orchestration tool that comports itself to be a job queue that we are too familiar with :roll_eyes: